### PR TITLE
Identity gate

### PR DESCRIPTION
**Pull request by mlodi-hqs**
Adds support for the `Identity` gate.

### CODE REVIEW

**Comment by mlodi-hqs** [comment]
Should I specify `Unreleased` in the CHANGELOG?

File: .compatibility_tests/compatibility_test_1_6/tests/integration/compatibility_1_6.rs [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+#[test_case(test_roqoqo_1_6::operations::ControlledRotateXY::new(0,1, 1.0.into(), 1.0.into()).into(); "ControlledRotateXY")]
+#[test_case(test_roqoqo_1_6::operations::ControlledControlledPauliZ::new(0,1,2).into(); "ControlledControlledPauliZ")]
+#[test_case(test_roqoqo_1_6::operations::ControlledControlledPhaseShift::new(0,1,2, 1.0.into()).into(); "ControlledControlledPhaseShift")]
+#[test_case(test_roqoqo_1_6::operations::Toffoli::new(0,1,2).into(); "Toffoli")]
+// 1.4
+#[test_case(test_roqoqo_1_6::operations::GPi::new(0, 0.1.into()).into(); "GPi")]
+#[test_case(test_roqoqo_1_6::operations::GPi2::new(0, 0.1.into()).into(); "GPi2")]
+// 1.5
+#[test_case(test_roqoqo_1_6::operations::PragmaControlledCircuit::new(0, test_roqoqo_1_6::Circuit::new()).into(); "PragmaControlledCircuit")]
+// Operations from 1.6 - uncomment for next unittests
```
Comment (kbarkhqs) [inline .compatibility_tests/compatibility_test_1_6/tests/integration/compatibility_1_6.rs:103]:
```suggestion
// Operations from 1.6
```

File: .compatibility_tests/compatibility_test_1_6/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+include = ["src*", "build.rs", "LICENSE", "README.md"]
+publish = false
+
+# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html
+
+[workspace]
+
+[dependencies]
+test_roqoqo_1_6 = {package = "roqoqo", version="=1.6.1"}
+test_roqoqo_derive_1_5 = {package = "roqoqo-derive", version="=1.6.1"}
```
Comment (kbarkhqs) [inline .compatibility_tests/compatibility_test_1_6/Cargo.toml:23]:
```suggestion
test_roqoqo_derive_1_6 = {package = "roqoqo-derive", version="=1.6.1"}
```

File: roqoqo/build.rs [hunk 3]
```diff
@@ -313,6 +313,9 @@ impl<'ast> Visit<'ast> for Visitor {
                 if trait_name.as_str() == "ImplementedIn1point6" {
                     self.roqoqo_version_register.insert(id.clone(), 6);
                 }
+                if trait_name.as_str() == "ImplementedIn1point7" {
```
Comment (kbarkhqs) [inline roqoqo/build.rs:316]: Is this all that needs adding for a new `ImplementedIn...` trait? I though there was more

File: qoqo/src/noise_models/error_on_gate.rs [hunk 4]
```diff
@@ -155,16 +155,13 @@ impl ErrorOnGateModelWrapper {
...
-                internal: noise_operator.clone(),
-            })
-        } else {
-            None
-        }
+            .map(
+                |noise_operator| struqture_py::spins::PlusMinusLindbladNoiseOperatorWrapper {
+                    internal: noise_operator.clone(),
+                },
+            )
```
Comment (kbarkhqs) [inline qoqo/src/noise_models/error_on_gate.rs:164]: Why has this file been changed?

File: roqoqo/build.rs (hunk 3 above)
Comment (mlodi-hqs) [reply roqoqo/build.rs:316]: I did too, but looking at the others that's all that is needed apparently.

File: qoqo/src/noise_models/error_on_gate.rs (hunk 4 above)
Comment (mlodi-hqs) [reply qoqo/src/noise_models/error_on_gate.rs:164]: clippy

**Review by kbarkhqs — APPROVED** [review]

**Review by nfwvogt — APPROVED** [review]

**Pull request merged**